nutanix/resource_nutanix_pbr.go

```diff
@@ -0,0 +1,634 @@
+	}
+
+	spec.Resources = res
+	request.Metadata = metadata
+	request.Spec = spec
+
+	// Make request to the API
+	resp, err := conn.V3.UpdatePBR(ctx, d.Id(), request)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	taskUUID := resp.Status.ExecutionContext.TaskUUID.(string)
+
+	// Wait for the PBR to be available
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"PENDING", "RUNNING"},
+		Target:     []string{"SUCCEEDED"},
+		Refresh:    taskStateRefreshFunc(conn, taskUUID),
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		Delay:      PbrDelayTime,
+		MinTimeout: PbrMinTimeout,
+	}
+
+	if _, errWaitTask := stateConf.WaitForStateContext(ctx); errWaitTask != nil {
+		return diag.Errorf("error waiting for PBR (%s) to create: %s", d.Id(), errWaitTask)
+	}
+
+	return resourceNutanixPbrRead(ctx, d, meta)
+}
+
+func resourceNutanixPbrDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*Client).API
+
+	log.Printf("[DEBUG] Deleting PBR: %s, %s", d.Get("name").(string), d.Id())
+	resp, err := conn.V3.DeletePBR(ctx, d.Id())
+	if err != nil {
+		if strings.Contains(fmt.Sprint(err), "ENTITY_NOT_FOUND") {
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("error while deleting PBR UUID(%s): %s", d.Id(), err)
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"DELETE_PENDING", "RUNNING"},
+		Target:     []string{"SUCCEEDED"},
+		Refresh:    taskStateRefreshFunc(conn, resp.Status.ExecutionContext.TaskUUID.(string)),
+		Timeout:    d.Timeout(schema.TimeoutDelete),
+		Delay:      PbrDelayTime,
+		MinTimeout: PbrMinTimeout,
+	}
+
+	if _, err := stateConf.WaitForStateContext(ctx); err != nil {
+		return diag.Errorf(
+			"error waiting for vpc (%s) to delete: %s", d.Id(), err)
+	}
+	d.SetId("")
+	return nil
+}
+
+func getPbrResources(d *schema.ResourceData, pbr *v3.PbrResources) error {
+	if sour, sok := d.GetOk("source"); sok {
+		pbr.Source = expandSourDest(sour.([]interface{}))
+	}
+
+	if dest, dok := d.GetOk("destination"); dok {
+		pbr.Destination = expandSourDest(dest.([]interface{}))
+	}
+
+	if proto, pok := d.GetOk("protocol_type"); pok {
+		pbr.ProtocolType = utils.StringPtr(proto.(string))
+	}
+
+	if priorty, prok := d.GetOk("priority"); prok {
+		pbr.Priority = utils.IntPtr(priorty.(int))
+	}
+
+	if bidirec, ok := d.GetOk("is_bidirectional"); ok {
+		pbr.IsBidirectional = utils.BoolPtr(bidirec.(bool))
+	}
+
+	if protoParam, pok := d.GetOk("protocol_parameters"); pok {
+		pbr.ProtocolParameters = expandProtocolParameters(protoParam.([]interface{}))
+	}
+
+	if action, aok := d.GetOk("action"); aok {
+		act := &v3.PbrAction{}
+
+		act.Action = utils.StringPtr(action.(string))
+
+		if sip, sok := d.GetOk("service_ip_list"); sok {
+			subips := sip.([]interface{})
+			sublist := make([]string, len(subips))
+			for a := range subips {
+				sublist[a] = subips[a].(string)
+			}
+			act.ServiceIPList = sublist
```

Comment:
Same. Check utils for this conversion functionality

Reply:
no util is present to convert list of strings to list of strings.